### PR TITLE
Introduction cleanup

### PR DESCRIPTION
**Pull request by mxstbr**
Took @KarandikarMihir's amazing work of art from #1366 and cleaned it up just a little to get it ready for the big release.

It's now called the "Hitchhikers Guide to `react-boilerplate`" to make sure we avoid having full-on React beginners think we'll teach them React. Other than that it's mostly cosmetic changes related to Markdown formatting, there wasn't anything left @KarandikarMihir didn't touch upon!

Super impressive, can't wait to show this to the world!

### CODE REVIEW

**Comment by KarandikarMihir** [comment]
@mxstbr Very concise and neat :sparkles: :100: Thank you!!

**Comment by mxstbr** [comment]
CI fails for the push because of "sh: 1: rimraf: not found" – that seems totally unrelated?

**Comment by Dattaya** [comment]
@mxstbr, yep, I've described what to do here: https://github.com/mxstbr/react-boilerplate/pull/1432#issuecomment-272398963

File: docs/general/introduction.md [hunk 1]
```diff
@@ -1,81 +1,101 @@
...
+
+> Opening an issue is the fastest way to draw the attention of the team, but please make it a point to read the [docs](https://github.com/mxstbr/react-boilerplate/tree/master/docs) and [contribution instructions](https://github.com/mxstbr/react-boilerplate/blob/master/.github/CONTRIBUTING.md) before you do. The issues section is specifically used for pointing out defects and suggesting enhancements, if you have a question about one of the tools please refer to StackOverflow instead.
 
 ## Tech Stack
-Here's a curated list of packages (in no particular order) that you should have in-depth knowledge of, before starting your awesome project. However, the best way to have a complete list of dependencies is to see [package.json](https://github.com/mxstbr/react-boilerplate/blob/master/package.json).
+
+Here's a curated list of packages (in no particular order) that you should have knowledge of, before starting your awesome project. However, the best way to have a complete list of dependencies is to see [package.json](https://github.com/mxstbr/react-boilerplate/blob/master/package.json).
```
Comment (KarandikarMihir) [inline docs/general/introduction.md]: > (in no particular order)

I know I wrote this but it seems unnecessary now that I read it. Also, logically if you don't know React, you won't understand react router and styled components. So the order does matter here. But it's not important enough to highlight.

**Pull request merged**